### PR TITLE
Namespace tag the disruptions.gauge metric

### PR DESCRIPTION
**Pull request by ptnapoleon**
## What does this PR do?

- [ ] Adds new functionality
- [ ] Alters existing functionality
- [ ] Fixes a bug
- [ ] Improves documentation or testing

Please briefly describe your changes as well as the motivation behind them:
- Users want a metric to see how many disruptions are running in their namespace. They can sort of do so with ongoing_duration, but that's a bit indirect. They should be able to check the disruptions.gauge and filter by a tag on their namespace.
- We don't currently tag `disruptions.gauge` at all. I've added a namespace tag

## Code Quality Checklist

- [x] The documentation is up to date.
- [x] My code is sufficiently commented and passes continuous integration checks.
- [x] I have signed my commit (see [Contributing Docs](../CONTRIBUTING.md)).

## Testing

- [ ] I leveraged continuous integration testing
    - [ ] by depending on existing `unit` tests or `end-to-end` tests.
    - [ ] by adding new `unit` tests or `end-to-end` tests.
- [ ] I manually tested the following steps:
    - `x`
    - [ ] locally.
    - [x] as a canary deployment to a cluster.

### CODE REVIEW

**Comment by ptnapoleon** [comment]
I've confirmed this works! the namespace tag for this metric now lists the number of disruptions in that namespace, no longer attributing them all to `chaos-engineering`

File: controllers/disruption_controller.go [hunk 1]
```diff
@@ -1043,15 +1045,29 @@ func (r *DisruptionReconciler) ReportMetrics(ctx context.Context) {
...
 		}
 
 		// send metrics
 		if err := r.MetricsSink.MetricStuckOnRemovalGauge(float64(stuckOnRemoval)); err != nil {
 			r.BaseLog.Errorw("error sending stuck_on_removal_total metric", "error", err)
 		}
 
-		if err := r.MetricsSink.MetricDisruptionsGauge(float64(len(l.Items))); err != nil {
-			r.BaseLog.Errorw("error sending disruptions.gauge metric", "error", err)
+		if len(namespaces) > 0 {
```
Comment (ptnapoleon) [inline controllers/disruption_controller.go:1061]: I've tested and this works in the case of 0 disruptions, disruptions in 1 namespace, and disruptions in many namespaces

File: controllers/disruption_controller.go [hunk 2]
```diff
@@ -1043,15 +1045,29 @@ func (r *DisruptionReconciler) ReportMetrics(ctx context.Context) {
 			chaosPodsCount += len(chaosPods)
 
 			r.handleMetricSinkError(r.MetricsSink.MetricDisruptionOngoingDuration(time.Since(d.ObjectMeta.CreationTimestamp.Time), []string{"disruptionName:" + d.Name, "namespace:" + d.Namespace}))
+
+			if count, ok := namespaces[d.Namespace]; ok {
+				namespaces[d.Namespace] = count + 1
```
Comment (zhiyanfoo) [inline controllers/disruption_controller.go:1050]:
```suggestion
			if +, ok := namespaces[d.Namespace]; ok {
				namespaces[d.Namespace] += 1
```

**Review by taihuynh167 — APPROVED** [review]

File: controllers/disruption_controller.go (hunk 1 above)
Comment (clairecng) [reply controllers/disruption_controller.go:1057]: Why do we need to make the verification here? If I'm not mistaken namespaces can't be nil no?

**Review by clairecng — APPROVED** [review]
👍

**Pull request merged**